server/channels/web/oauth.go

```diff
@@ -625,3 +628,54 @@ func getAuthorizationServerMetadata(c *Context, w http.ResponseWriter, r *http.R
 		c.Logger.Warn("Error writing authorization server metadata response", mlog.Err(err))
 	}
 }
+
+// loginWithIntune handles authentication using Microsoft Intune MAM with Entra ID tokens
+func loginWithIntune(c *Context, w http.ResponseWriter, r *http.Request) {
+	var req model.IntuneLoginRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		c.SetInvalidParamWithErr("request_body", err)
+		return
+	}
+
+	if req.AccessToken == "" {
+		c.SetInvalidParam("access_token")
+		return
+	}
+
+	auditRec := c.MakeAuditRecord("login_with_intune", model.AuditStatusFail)
+	defer c.LogAuditRec(auditRec)
+	c.LogAudit("attempt")
+
+	// Authenticate user via Intune MAM
+	user, err := c.App.LoginByIntune(c.AppContext, req.AccessToken)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	auditRec.AddMeta("obtained_user_id", user.Id)
+	c.LogAuditWithUserId(user.Id, "obtained user")
+
+	isMobile := req.DeviceId != ""
+	session, err := c.App.DoLogin(c.AppContext, w, r, user, req.DeviceId, isMobile, true, false)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	c.AppContext = c.AppContext.WithSession(session)
+
+	w.Header().Set(model.HeaderToken, session.Token)
```

Comment:
This is not needed here as `c.App.DoLogin()` already sets this: https://github.com/mattermost/mattermost/blob/master/server/channels/app/login.go#L191